lms/djangoapps/courseware/views/views.py

```diff
@@ -765,7 +765,7 @@ def course_about(request, course_id):
         course_details = CourseDetails.populate(course)
         modes = CourseMode.modes_for_course_dict(course_key)
 
-        if configuration_helpers.get_value('ENABLE_MKTG_SITE', settings.FEATURES.get('ENABLE_MKTG_SITE', False)):
+        if not has_access(request.user, 'load', course) and configuration_helpers.get_value('ENABLE_MKTG_SITE', settings.FEATURES.get('ENABLE_MKTG_SITE', False)):
             return redirect(reverse(course_home_url_name(course.id), args=[text_type(course.id)]))
```

Comment:
so from what I understand here is that courseware home is accessible even if your not enrolled but not the other way around, am I right? How does this line prevent the cyclic redirects?

Reply:
what happened is the QA copied the course link 

`/courses/course-v1:ORG1ORG1+123+2020_T1/course/`

then went the spec that is the parent to this course and un-enrolled from it then she pasted the link into the browser, then this function `course_about` [`lms/djangoapps/courseware/views/views.py`]  got activated and because of this line 

```py
configuration_helpers.get_value('ENABLE_MKTG_SITE', settings.FEATURES.get('ENABLE_MKTG_SITE', False))
```

the user got redirected to `/courses/course-v1:ORG1ORG1+123+2020_T1/course/` and since she is not enrolled again she got redirected to the about page and got stuck in an infinite loop 


----

thinking about it, I don't think this is the best solution, but what this the purpose behind this line?

```
configuration_helpers.get_value('ENABLE_MKTG_SITE', settings.FEATURES.get('ENABLE_MKTG_SITE', False))
```

it's an old commit https://github.com/Edraak/edraak-platform/commit/8ae92901ef882ee18323eb87447f4090a0098716 

before 4 years

Reply:
This issue is even in the old dashboard, so I think it's old

Reply:
Jenkins build fails, but I can't see why 

probably a test